docs/pages/application-access/getting-started.mdx

```diff
@@ -23,38 +36,42 @@ Let's connect to Grafana using Teleport in three steps:
 
 - (!docs/pages/includes/tctl.mdx!)
 - A host where you will run the Teleport Application Service.
+- A DNS subdomain for each application.
+   
+   <Admonition type="note" title="Application Access and DNS for Self-Hosted Clusters">
 
-<Tabs>
-<TabItem label="Docker">
-- A Docker installation, which we will use to launch Grafana in a container.
-  Alternatively, if you have another web application you'd like to protect with
-  Teleport, you can use that instead.
-</TabItem>
-<TabItem label="Kubernetes Cluster">
+   (!docs/pages/includes/dns-app-access.mdx!)
 
-(!docs/pages/includes/kubernetes-access/helm-k8s.mdx!)
+   Teleport Cloud will automatically provide a subdomain and signed TLS certificate for your application under your tenant address.
+   </Admonition>
 
-- Administrative rights to deploy within a cluster.
+- Docker or Kubernetes to launch Grafana.
 
-</TabItem>
-</Tabs>
+   This tutorial illustrates how to launch Grafana in a Docker container or Kubernetes cluster.
+   Therefore, you must have either Docker or Kubernetes installed. For more information, click **Docker** or **Kubernetes cluster**.
+   
+   <Tabs>
+   <TabItem label="Docker">
+   You must have some version of Docker installed to launch Grafana in a Docker container. 
+   If you are configuring access to a different web application using Teleport, use the appropriate commands to 
+   launch that application in a Docker container.
+   </TabItem>
+   <TabItem label="Kubernetes cluster">
 
-<Admonition type="tip" title="Not yet a Teleport user?">
-{/*lint ignore messaging*/}
-If you have not yet deployed the Auth Service and Proxy Service, you should
-follow one of our [getting started guides](../get-started.mdx) or try our
-Teleport Application Access [interactive learning track](https://play.instruqt.com/teleport/invite/rgvuva4gzkon).
-</Admonition>
+   (!docs/pages/includes/kubernetes-access/helm-k8s.mdx!)
+
+   - Administrative rights to deploy within a cluster.
 
-<Admonition type="note" title="Application Access and DNS for Self-Hosted Clusters">
+   </TabItem>
+   </Tabs>
 
-(!docs/pages/includes/dns-app-access.mdx!)
+- Check that you are using appropriate permissions.
+   The permissions used in examples in this tutorial are intended for demo environments.
+   Before running Teleport in a production environment, check that you are following best practices to avoid security incidents.
 
-Teleport Cloud will automatically provide a subdomain and signed TLS certificate
-for your application under your tenant address.
-</Admonition>
+   (!docs/pages/includes/permission-warning.mdx!)
 
-## Step 1/4. Generate a token
+## Step 1/5. Generate a token
 
 A join token is required to authorize a Teleport Application Service instance to
 join the cluster. Generate a short-lived join token and save it, for example,
```

Comment:
While we're in here, we should specify that the join token is being saved on the host where we will run the Teleport Application Service.

Reply:
I think the instructions are a little vague here. I took a stab at making them somewhat more specific. It seems like I needed to run this particular command on the instance where the Auth service runs. To run it on an application service host seems like it would require additional command-line options, like --identity and --auth-server, but I'm not sure how you create an identity file without digging in deeper, so I modified the description around this a little.

Reply:
The `tctl` command can be run on the auth service or the local workstation after logging in with `tsh`. We should specify the latter. The point I'm making is that the file we save the token string to should be on the application service host, so that when we start that service it can read the token.